src/client/sandbox/event/simulator.js

```diff
@@ -316,24 +327,84 @@ export default class EventSimulator {
     }
 
     _dispatchKeyEvent (el, args) {
-        var ev = null;
+        var ev                        = null;
+        var browserWithNewEventsStyle = !browserUtils.isIE || browserUtils.version > 11;
 
-        if (nativeMethods.documentCreateEvent) {
-            ev = nativeMethods.documentCreateEvent.call(document, 'Events');
-            ev.initEvent(args.type, args.canBubble, args.cancelable);
-            ev = extend(ev, {
-                view:     args.view,
-                detail:   args.detail,
-                ctrlKey:  args.ctrlKey,
-                altKey:   args.altKey,
-                shiftKey: args.shiftKey,
-                metaKey:  args.metaKey,
-                keyCode:  args.keyCode,
-                charCode: args.charCode,
-                which:    args.which
+        if (browserWithNewEventsStyle && nativeMethods.WindowKeyboardEvent) {
+            ev = new nativeMethods.WindowKeyboardEvent(args.type, {
+                bubbles:          args.canBubble,
+                cancelable:       args.cancelable,
+                cancelBubble:     false,
+                defaultPrevented: false,
+                view:             args.view,
+                detail:           args.detail,
+                ctrlKey:          args.ctrlKey,
+                altKey:           args.altKey,
+                shiftKey:         args.shiftKey,
+                metaKey:          args.metaKey,
+                keyCode:          args.keyCode,
+                charCode:         args.charCode,
+                which:            args.which
+            });
+        }
+        else if (nativeMethods.documentCreateEvent) {
+            ev = nativeMethods.documentCreateEvent.call(document, 'KeyboardEvent');
+
+            ev.initKeyboardEvent(args.type, args.canBubble, args.cancelable, args.view, '', 0, EventSimulator._getModifiersAsString(args), false, '');
+        }
+
+        if (ev) {
+            // NOTE: keyCode, charCode and which properties are not assigned after creation of KeyboardEvent
+            Object.defineProperty(ev, 'keyCode', {
+                get: () => args.keyCode
+            });
+
+            Object.defineProperty(ev, 'charCode', {
+                get: () => args.charCode
+            });
+
+            Object.defineProperty(ev, 'which', {
+                get: () => args.which
             });
 
-            return this._raiseDispatchEvent(el, ev, args);
+            var prevented   = false;
+            var returnValue = true;
+
+            // NOTE: dispatchEvent method does not return false in case of preventDefault was
```

Comment:
// NOTE: the dispatchEvent method does not return false in the case when preventDefault was 
// raised for events that were created with the KeyboardEvent constructor